staging/src/k8s.io/apimachinery/pkg/util/wait/wait.go

```diff
@@ -37,12 +37,34 @@ var ForeverTestTimeout = time.Second * 30
 // NeverStop may be passed to Until to make it never stop.
 var NeverStop <-chan struct{} = make(chan struct{})
 
-// StartUntil starts f in a new goroutine and calls done once f has finished.
-func StartUntil(stopCh <-chan struct{}, wg *sync.WaitGroup, f func(stopCh <-chan struct{})) {
-	wg.Add(1)
-	go func() {
-		defer wg.Done()
+// Group is an interface to decouple code from sync.WaitGroup.
+type Group interface {
+	Add(delta int)
+	Done()
+}
+
+// StartWithChannelWithinGroup adds 1 to the group, starts f in a new goroutine and calls g.Done once f has finished.
+// stopCh is passed to f as an argument.
+func StartWithChannelWithinGroup(stopCh <-chan struct{}, g Group, f func(stopCh <-chan struct{})) {
```

Comment:
Right. I think mixing `wait.Group` with `sync.WaitGroup` should not happen - it may cause confusion and bugs (because of misuse). However when there is a need, `wg` field can always be made public to wrap existing `sync.WaitGroup`. Also one can always do what those methods do.